### PR TITLE
console history persistence

### PR DESCRIPTION
**Pull request by kasjer**
Console history now can be kept in log as well as in ram.

### CODE REVIEW

File: sys/console/full/history_log/src/history_log.c [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+#endif
+#if MYNEWT_VAL(LOG_FCB2)
+#include <fcb/fcb2.h>
+#endif
+#include <log/log.h>
+#include <console/history.h>
+
+#define HISTORY_CACHE_SIZE MYNEWT_VAL(CONSOLE_HISTORY_LOG_CACHE_SIZE)
+
+#define VARIANT 2
```
Comment (andrzej-kaczmarek) [inline sys/console/full/history_log/src/history_log.c]: some leftover?

File: sys/console/full/syscfg.yml [hunk 2]
```diff
@@ -43,15 +43,31 @@ syscfg.defs:
...
         description: >
-            Number of lines to be stored in console history.
-            Set to "0" to disable console history.
+            This is no longer used.
+            Equivalent settings:
+                CONSOLE_HISTORY: ram
+                CONSOLE_HISTORY_RAM_HISTORY_SIZE: <n>   <- previous value of CONSOLE_HISTORY_SIZE
         value: 0
-    CONSOLE_MAX_PROMPT_LEN:
+        deprecated: 1
```
Comment (andrzej-kaczmarek) [inline sys/console/full/syscfg.yml]: if this is not used (i.e. it does nothing as there is no new syscfg which "inherits" its value) then it should better be `defunct`. also I'd move it somewhere at the end of settings so it does not bother anyone

File: sys/console/full/syscfg.yml [hunk 3]
```diff
@@ -43,15 +43,31 @@ syscfg.defs:
...
+    CONSOLE_HISTORY_AUTO_SEARCH:
+        description: >
+            When set to 1 history is automatically searched for lines starting with already typed
+            text.
+        value: 0
+    CONSOLE_PROMPT_MAX_LEN:
         description: 'Maximum number of characters for prompt'
         value: 16
-    CONSOLE_STICKY_PROMPT:
+    CONSOLE_PROMPT_PERSIST:
```
Comment (andrzej-kaczmarek) [inline sys/console/full/syscfg.yml]: `STICKY` was nicer :)

File: sys/console/full/history_log/src/history_log.c (hunk 1 above)
Comment (kasjer) [reply sys/console/full/history_log/src/history_log.c]: indeed, removed

File: sys/console/full/syscfg.yml (hunk 3 above)
Comment (kasjer) [reply sys/console/full/syscfg.yml]: switched to CONSOLE_PROMPT_STICKY

File: sys/console/full/syscfg.yml (hunk 2 above)
Comment (kasjer) [reply sys/console/full/syscfg.yml]: moved to end and changed to defunct

**Pull request merged**